### ISSUE TITLE
lenght and width measurements

### ISSUE THREAD

**Issue by Daphnisd** [iobis/vocab#7]
At the moment we have for length the general terms:
http://seadatanet.maris2.nl/bandit/browse_step.php?&step=009BE007117_006S0600015

We also have specific lengths: 
http://seadatanet.maris2.nl/bandit/browse_step.php?&step=009BE007117_001length_012MAT00906

I would like to have 3 sets (maybe created upon need) of parameters for each length type.
1) specimen length (referring to a single specimen / a single measurement)
2) average length (so the value provided is the average for x number of specimens
3) min length 
4) max length
The min and the max lenght are needed because abundances are often given for a size class. The min and max values allow you to specify size classes.

It's unclear what the best naming strategy would be for each of these 4 types of length parameters. I propose we decide on best practice for names using the general length parameters (for which all 4 are needed) and expand this upon need to other measurement terms. @gwemon please feel free to add or correct options  (I don't think I recorded all possibilities).

**Option 1)**
1) Specimen length:   
- Length (specimen) of biological entity specified elsewhere 

2) average length 
- Length (specimen mean) of biological entity specified elsewhere 

3) min length 
- Length (specimen minimum) of biological entity specified elsewhere 

4) max length 
- Length (specimen maximum) of biological entity specified elsewhere 

**Option 2)**
1) Specimen length:   
- Length of biological entity specified elsewhere 

2) average length 
- Length (mean) of biological entity specified elsewhere 

3) min length 
- Length (minimum) of biological entity specified elsewhere 

4) max length 
- Length (maximum) of biological entity specified elsewhere 

Actually length itself might not be clear enough.  It think it the "total length" which is meant (anyway the maxinum length in the way that it's commonly measured for that that taxon ). Should total then be included in the parameter name (I actually think it should), but then you have:

**Option 3)**
1) Specimen length:   
- Length (total) of biological entity specified elsewhere 

2) average length 
- Length (total mean) of biological entity specified elsewhere 

3) min length 
- Length (total minimum) of biological entity specified elsewhere 

4) max length 
- Length (total maximum) of biological entity specified elsewhere 

Lastly: in a dataset i have a parameter called "carapace width". I guess that this will be the same as "total width", so should a parameter for carapace width be created or should a total width parameter be used? At the moment we still don't have any width parameters in the P01 vocabulary.

**Comment by OBISCanada** [comment]
Just a quick question → I assume that the count is stored elsewhere, correct?
If yes then #1 could be revised to simply specimen length (single size measurement).  There could be many specimens of this one size.
Also – any need to store the standard deviation if there is an average? (overkill?)

Should part of developing datatypes be considering possible data?
Some datasets will have a long list of sizes: 10,11,10,12,13,15,10,11, ...
Other datasets will have frequencies: 10-3; 11-2;12-1; 13-1,14-0; 15-1
Others may have size classes: 9.5-10.49(3); 10.5-11.49(2)
People often forget to include the precision of their measurements... did they measure to the nearest mm? In other words there are usually min/max values associated with a measurement.


Sent from Mail for Windows 10

From: Daphnis De Pooter
Sent: November 29, 2018 12:42 PM
To: iobis/vocab
Cc: Subscribed
Subject: [iobis/vocab] lenght and width measurements (#7)

At the moment we have for length the general terms:
http://seadatanet.maris2.nl/bandit/browse_step.php?&step=009BE007117_006S0600015
We also have specific lengths:
http://seadatanet.maris2.nl/bandit/browse_step.php?&step=009BE007117_001length_012MAT00906
I would like to have 3 sets (maybe created upon need) of parameters for each length type.
1. specimen length (referring to a single specimen / a single measurement)
2. average length (so the value provided is the average for x number of specimens
3. min length
4. max length
The min and the max lenght are needed because abundances are often given for a size class. The min and max values allow you to specify size classes.
It's unclear what the best naming strategy would be for each of these 4 types of length parameters. I propose we decide on best practice for names using the general length parameters (for which all 4 are needed) and expand this upon need to other measurement terms. @gwemon please feel free to add or correct options (I don't think I recorded all possibilities).
Option 1)
1. Specimen length:
• Length (specimen) of biological entity specified elsewhere
2. average length
• Length (specimen mean) of biological entity specified elsewhere
3. min length
• Length (specimen minimum) of biological entity specified elsewhere
4. max length
• Length (specimen maximum) of biological entity specified elsewhere
Option 2)
1. Specimen length:
• Length of biological entity specified elsewhere
2. average length
• Length (mean) of biological entity specified elsewhere
3. min length
• Length (minimum) of biological entity specified elsewhere
4. max length
• Length (maximum) of biological entity specified elsewhere
Actually length itself might not be clear enough. It think it the "total length" which is meant (anyway the maxinum length in the way that it's commonly measured for that that taxon ). Should total then be included in the parameter name (I actually think it should), but then you have:
Option 3)
1. Specimen length:
• Length (total) of biological entity specified elsewhere
2. average length
• Length (total mean) of biological entity specified elsewhere
3. min length
• Length (total minimum) of biological entity specified elsewhere
4. max length
• Length (total maximum) of biological entity specified elsewhere
Lastly: in a dataset i have a parameter called "carapace width". I guess that this will be the same as "total width", so should a parameter for carapace width be created or should a total width parameter be used? At the moment we still don't have any width parameters in the P01 vocabulary.

**Comment by Daphnisd** [comment]
Good point, we may want to say 5 specimens have this length and I agree that the same term to specify the length of a single specimen should be used.

Standard deviation: It's probably relevant and I think it might occur in a dataset (but I would create terms upon need).

Precision can be dealt with by dwc:measurementAccuracy (http://rs.tdwg.org/dwc/terms/#measurementAccuracy)?

**Comment by gwemon** [comment]
Am I correct to understand that for the minimum and maximum body lengths (no matter how they are defined) we need two kind of concepts: the minimum and maximum length of the size classes within a biometric distribution and the minimum and maximum length of individuals within a population? Apologies if that was mentioned earlier and I missed it.

**Comment by gwemon** [comment]
Do we know good sources of biometrics terminology that we could use as reference?
I agree with @Daphnisd comment under #1 that we may need to review what we've done up till now and use specific terminology that accurately define what is being reported.
So for example if what is being reported is the total length of an individual specimen then the P01 term should be "Total length (individual specimen) of biological entity specified elsewhere" -- terms along these lines;
If what is reported is the average total length of individuals within a population then you could have "Total length (population) mean of b.e.s.e" -- is "population" the right term? shall we use "Total length (in assayed sample)" in keeping with what has been used for, for example "Count (in assayed sample)?

At the moment I envisage "Total length (individual specimen)" and "Total length (population)" to be defined in https://www.bodc.ac.uk/resources/vocabularies/vocabulary_search/S06/; 
"mean", "standard deviation", "minimum", "maximum" etc... are already defined in S07.

We can then look at how to deal with the minimum and maximum in size-classes as used in frequency distribution. I'll check to see how we manage this for other kind of frequency distributions eg particle size-distributions.

Finally we can look at the various length and width measurements used for different types of organisms. If we feel that these are simply different biometric parameter entities then they would fit in S06 as eg "Culmen length (individual specimen)", "Carapace length (individual specimen)", Carapace length (population) mean, "Culmen length (population) mean" etc. One thing that makes me feel this is the way to go is that if one googles "Carapace length" the first 3 hits are all related to standard measurement techniques for sea turtles... if one googles "Culmen length" the top hits are related to birds biometrics. So by keeping these terms as they are we make it easier to align to standard or well used terminology. So bearing that in mind, I don't think the suggestion I made in issue #1 of using the subcomponent field for the body part would make sense for biometric parameters.

**Comment by Daphnisd** [comment]
Some scenario's I can think of:

#### Data from single sampling event. Given = average, min and max "total lengths"
a) Total length (population) mean
b) Total length (population) minimum
c) Total length (population) maximum

I'm not sure if data from a single sample can be called "population"? If use "Total length (population) maximum" I would expect the value to be the the maximum lenght that a specimen can get in that area. But it's actually just the max length in my sample. 

#### Data from single sampling event. Given = length frequencies"
a) "Total length (population)" +  "count of occurrence" for these length measurements

#### Data from 2 single specimen.
a) Total length (individual specimen)    
b) Total length (individual specimen) 

#### Length classes. Given:  (length "2-4" = abundance x),  (length "4-6" = abundance y)
occurrence record 1
a) Total length (individual specimen) minimum =  2
b) Total length (individual specimen) maximum = 4
c) abundance = X
occurrence record 2
a) Total length (individual specimen) minimum =  4
b) Total length (individual specimen) maximum = 6
c) abundance = y

Or would we for length classes simply use "Total length minimum"? Or is this still population? ("it doesn't seem right to use here "individual specimen")
 



I agree with formatting like "Carapace length (individual specimen)" and "Culmen length  (individual specimen)".

**Comment by Daphnisd** [comment]
> 
> 
> Am I correct to understand that for the minimum and maximum body lengths (no matter how they are defined) we need two kind of concepts: the minimum and maximum length of the size classes within a biometric distribution and the minimum and maximum length of individuals within a population? Apologies if that was mentioned earlier and I missed it.

I'm actually not sure on this... are they really different things? It's about the minimum and maximum in a sample and the minimum and the maximum of a subset (size class) of the sample.

**Comment by OBISCanada** [comment]
population = group observed during an event (location/date)

Get rid of population and individuals and replace with ‘specimen(s)’

You need to have flexibility so that one can add individual size measurements OR add group size measurements (frequencies).  If I measured 10 individuals I might enter 10 entries. If I measured 100 individuals I would enter frequencies.

I assume that there is a field ‘ perhaps called organismID or individualID or something similar that can link measurements of an individual or group of individuals. Example I may have measured length, width and weight of one individual or a group of individuals.

**Comment by gwemon** [comment]
> population = group observed during an event (location/date) Get rid of population and individuals and replace with ‘specimen(s)’

Do you mean to replace "Total length (population)" or "Total length (individual organism)" by "Specimen total length"? 

You'd then have the P01 concept:
Specimen total length of biological entity spec elsewhere - fine, this is clearly the length of a specific individual measured at a specific time/location.
Specimen total length mean of biological entity specified elsewhere - this for me is not clear: does this refer to the population average or the individual's average length? Maybe this is unlikely to happen for length unless an organism has his length measured several times and given as an average, but what's about weight or width? these can realistically be given as an average for one individual... I really think that we need to distinguish these 2 kinds of measurements.

**Comment by gwemon** [comment]
> I'm actually not sure on this... are they really different things? It's about the minimum and maximum in a sample and the minimum and the maximum of a subset (size class) of the sample.

Yes for me these are 2 different parameter concepts: one is the total length min/max of an organism or multiple organisms within a population, the other one is the lower and upper limit of one of multiple size-class ranges in a frequency distribution.

**Comment by OBISCanada** [comment]
How can one individual have multiple length measurements?  If you measure several times for some reason then these must be different measurements – correct?  You can measure me several times but I am only one height.

Sent from Mail for Windows 10

**Comment by OBISCanada** [comment]
An organism can only have one measurement. You can treat this as min/max but that is based on the measuring method – accurate to unit.

I don’t think that I am helping with this discussion.
Maybe it would work best  with a number of real examples.
Not everyone will record their measurements the same way!

Sent from Mail for Windows 10

**Comment by Daphnisd** [comment]
I'm trying to summarize again:

Do we have consensus that we need codes to deal with the following situations?
```
a. 1 code to refer to a single specimen
b. 1 code for an average (of a population)
c. 2 codes for min/max of population 
d. 2 codes for min/max size class
e. 1 code for a single measurement which applies to different specimens in the same sample
```

So what I think codes would look like:
**1) Specimen option**
> a + e)  Specimen total length of biological entity specified elsewhere
> b) Specimen total length (mean) of biological entity specified elsewhere
> c + d) Specimen total length (minimum) of biological entity specified elsewhere  / Specimen total length (maximum ) of biological entity specified elsewhere

Problems: No possibility to create terms referring to the average measurement of a single individual and b may be unclear to some users. I don't think this is a big issue for OBIS (but maybe elsewhere). 
I added size classes with the population minimums but for me they can be split up too (Size class total length (minimum)?).

**2) Population / individual organism option**

> a. Total length (individual organism) of biological entity specified elsewhere
> b. Population total length (mean) of biological entity specified elsewhere
> c. Population total length (minimum) of biological entity specified elsewhere / Population total length (maximum) of biological entity specified elsewhere
> d. Size class total length (minimum) of biological entity specified elsewhere / Size class total length (maximum) of biological entity specified elsewhere
> e. Population total length (individual organisms) of biological entity specified elsewhere

the naming of d and e are maybe not perfect yet



I like 1 cause there are less codes (4 vs 7) which will make picking the right one easier. I like 2 because terms are more specific and clear and and may be more sustainable on the long run.
